scripts/map_tokens.py

```diff
@@ -1,23 +1,50 @@
 #!/usr/bin/env python3
 
 import argparse
-import pickle
+
+import pandas as pd
+from datasets import Dataset
 
 from delphi.constants import STATIC_ASSETS_DIR
 from delphi.eval.token_map import token_map
 from delphi.eval.utils import load_validation_dataset
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="")
+
+    parser.add_argument(
+        "dataset_name",
+        type=str,
+        help="Dataset from huggingface to run token_map on",
+    )
+    parser.add_argument(
+        "--username",
+        type=str,
+        help="Hugging Face API username",
+    )
     parser.add_argument(
-        "dataset_name", help="Dataset from huggingface to run token_map on"
+        "--token",
+        type=str,
+        help="Hugging Face API token",
     )
-    parser.add_argument("--output", help="Output file name", default="token_map.pkl")
     args = parser.parse_args()
 
     dataset = load_validation_dataset(args.dataset_name)
 
-    mapping = token_map(dataset)
+    mapping = token_map(
+        dataset
+    )  # outputs the dictionary: dict[int, list[tuple[int, int]]]
 
-    with open(f"{STATIC_ASSETS_DIR}/{args.output}", "wb") as f:
-        pickle.dump(mapping, file=f)
+    complete_mapping = {key: mapping.get(key, None) for key in range(4096)}
+
+    hf_dataset = Dataset.from_dict({"prompt_pos_idx": list(complete_mapping.values())})
```

Comment:
Are we guaranteed to maintain ordering here?

Reply:
Seems like it might be better for this to be a list and not use a dictionary at all, since ordering is going to be deterministic and indices are ordered integers.